File: back/models/nutrition.py
from pydantic import BaseModel
from typing import List, Optional

class NutritionProfile(BaseModel):
    goal: str
    weight: float
    height: float
    age: int
    sex: str
    activityLevel: str
    allergies: List[str] = []

class Meal(BaseModel):
    type: str
    name: str
    kcal: int
    macros: str

class HistoryDay(BaseModel):
    date: str
    calories: int
    target: int
    status: str

class FoodItem(BaseModel):
    id: int # Mantuvimos IDs numéricos en el seed para compatibilidad
    name: str
    detail: str

class FoodAnalysis(BaseModel):
    is_food: bool
    calories: Optional[int] = None
    protein: Optional[int] = None
    fat: Optional[int] = None
    message: Optional[str] = None

class Notification(BaseModel):
    id: int
    type: str
    title: str
    time: str
    description: str
    isRead: bool

class Appointment(BaseModel):
    date: str
    time: str
    type: str

File: front/src/models/nutrition.ts
// This file contains TypeScript interfaces that correspond to the Pydantic models in the backend.

export interface NutritionProfile {
  goal: string;
  weight: number | string; // string to allow empty input field
  height: number | string;
  age: number | string;
  sex: string;
  activityLevel: string;
  allergies: string[];
}

export interface Meal {
  type: string;
  name: string;
  kcal: number;
  macros: string;
}

export interface HistoryDay {
  date: string;
  calories: number;
  target: number;
  status: 'inprogress' | 'success' | 'warning';
}

export interface FoodItem {
  id: number;
  name: string;
  detail: string;
}

export interface FoodAnalysis {
  is_food: boolean;
  calories?: number;
  protein?: number;
  fat?: number;
  message?: string;
}

export interface Notification {
  id: number;
  type: string;
  title: string;
  time: string;
  description: string;
  isRead: boolean;
}

export interface Appointment {
  date: string;
  time: string;
  type: string;
}

